### DISCUSSION TITLE
Why is GitHub notifying me about Homebrew's auth token?

### DISCUSSION THREAD

**Discussion by slhck** [Homebrew/discussions#1554] [Everyday usage] [answered]
I just got this message from GitHub:

> We noticed your personal access token, XXXXXX, has an outdated format and was used to access the GitHub API on May 27th, 2021 at 08:13 (UTC) with a user-agent header of Homebrew/3.1.9-56-gae8a936 (Macintosh; Intel Mac OS X 10.15.7) curl/7.64.1.
>
> We recently updated the format of our API authentication tokens, providing additional security benefits to all our customers.
>
> In order to benefit from this new format, please regenerate your personal access token, satyr, using the button below.

Why did I get this, and why would Homebrew authenticate with GitHub? I thought it – if at all – was using public APIs that do not require authentication?

All I did prior to this was running `brew tap` for a public tap and `brew install` from that tap.

Note that I haven't set any env variable specifying that token, so I'm unsure from where it was sourced.

Should I regenerate the token from GitHub's side, and what happens if I do that? Will it break Homebrew?

Perhaps there should be a better message for end-users on what to do in this case.

**Comment by SMillerDev** [comment]
What does `brew config` say?

**Reply by slhck** [reply]
```
HOMEBREW_VERSION: 3.1.9-56-gae8a936
ORIGIN: https://github.com/Homebrew/brew
HEAD: ae8a9368b9196560dbf399aed409b44a054414ef
Last commit: 3 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 394d44daf72f897476d21f7b0cb5d35a6905e557
Core tap last commit: 35 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: vim
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: octa-core 64-bit kabylake
Clang: 12.0.0 build 1200
Git: 2.31.1 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 10.15.7-x86_64
CLT: 12.0.0.32.29
Xcode: 12.4
```

**Reply by SMillerDev** [reply]
That's odd pinging: @Homebrew/brew 

Will brew use GitHub tokens from non `HOMEBREW_GITHUB_API_TOKEN` sources?

**Reply by nandahkrishna** [reply]
We do check the keychain for GitHub API tokens: [`api.rb`](https://github.com/Homebrew/brew/blob/ae8a9368b9196560dbf399aed409b44a054414ef/Library/Homebrew/utils/github/api.rb#L118). Setting this is necessary now with password-based authentication being deprecated.

**Reply by slhck** [reply]
I only have this in my keychain:

![image](https://user-images.githubusercontent.com/582444/119802624-fd806780-bede-11eb-98b4-280767ad7916.png)

Is that the one you are referring to?

**Reply by nandahkrishna** [reply]
That is the one, yes. To answer another question from your original post, Homebrew does need authentication with your PAT for some situations – when you're using `bump-formula-pr` or `bump-cask-pr`, for example. In your case, running `brew tap` might have used these credentials: [see here](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/about-remote-repositories#cloning-with-https-urls).

**Comment by nandahkrishna** [comment, marked as answer]
> Should I regenerate the token from GitHub's side, and what happens if I do that? Will it break Homebrew?

You might want to regenerate your PAT (and update your credentials in Keychain as well), just as GitHub suggests. Not doing so won't break Homebrew or any of your interactions with GitHub at the moment, but it may in the future (if GitHub decides to prevent API requests made with outdated tokens).

These links may be helpful: [[1]](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token), [[2]](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/about-remote-repositories) and [[3]](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git/updating-credentials-from-the-macos-keychain).

**Reply by slhck** [reply]
Thank you!
